src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -357,20 +369,47 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
       withWorkspace(mcnp.source.workspaceName.namespace, mcnp.source.workspaceName.name, txn) { srcWorkspace =>
         methodConfigurationDAO.get(mcnp.source.workspaceName.namespace, mcnp.source.workspaceName.name, mcnp.source.namespace, mcnp.source.name, txn) match {
           case None => RequestComplete(StatusCodes.NotFound)
-          case Some(methodConfig) =>
-            withWorkspace(mcnp.destination.workspaceName.namespace, mcnp.destination.workspaceName.name, txn) { destWorkspace =>
-              methodConfigurationDAO.get(mcnp.destination.workspaceName.namespace, mcnp.destination.workspaceName.name, mcnp.destination.namespace, mcnp.destination.name, txn) match {
-                case Some(existingMethodConfig) => RequestComplete(StatusCodes.Conflict,existingMethodConfig)
-                case None =>
-                  val target = methodConfig.copy(name = mcnp.destination.name, namespace = mcnp.destination.namespace, workspaceName = mcnp.destination.workspaceName)
-                  val targetMethodConfig = methodConfigurationDAO.save(target.workspaceName.namespace, target.workspaceName.name, target, txn)
-                  RequestComplete(StatusCodes.Created, targetMethodConfig)
-            }
-          }
+          case Some(methodConfig) => saveCopiedMethodConfiguration(methodConfig, mcnp.destination, txn)
         }
       }
     }
 
+  def copyMethodConfigurationFromMethodRepo(methodRepoQuery: MethodRepoConfigurationQuery, authCookie: HttpCookie): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      val url = Seq(methodRepoBaseURL, "configurations", methodRepoQuery.methodRepoNamespace, methodRepoQuery.methodRepoName, methodRepoQuery.methodRepoSnapshotId) mkString "/"
+
+      import scala.concurrent.ExecutionContext.Implicits.global
+      val pipeline = addHeader(Cookie(authCookie)) ~> sendReceive ~> unmarshal[AgoraEntity]
+      Try {
+        Await.result(pipeline(Get(url)), atMost = Duration(5, TimeUnit.SECONDS))
+      } match {
+        case Failure(error) =>
+          RequestComplete(StatusCodes.NotFound, error.getMessage)
```

Comment:
will this error always mean not found? I think we should pass through any error status from agora.